data/World/Dodongos Cavern MQ.json

```diff
@@ -42,7 +42,61 @@
                 can_play(Song_of_Time) and (can_child_attack or is_adult)",
             "Dodongos Cavern MQ GS Larvae Room": "can_use(Sticks) or has_fire_source",
             "Dodongos Cavern MQ GS Lizalfos Room": "can_blast_or_smash",
-            "Dodongos Cavern MQ Deku Scrub Staircase": "can_stun_deku"
+            "Dodongos Cavern MQ Deku Scrub Staircase": "can_stun_deku",
+            "Dodongos Cavern MQ Torch Puzzle Room Recovery Heart": "can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire)",
+            "Dodongos Cavern MQ Staircase Pot 1": "True",
+            "Dodongos Cavern MQ Staircase Pot 2": "True",
+            "Dodongos Cavern MQ Staircase Pot 3": "True",
+            "Dodongos Cavern MQ Staircase Pot 4": "True",
+            "Dodongos Cavern MQ Upper Lizalfos Pot 1": "
+                can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire) or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Upper Lizalfos Pot 2": "
+                can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire) or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Upper Lizalfos Pot 3": "
+                can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire) or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Upper Lizalfos Pot 4": "
+                can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire) or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Torch Puzzle Room Pot 1": "
+                can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire) or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Torch Puzzle Room Pot 2": "
+                can_blast_or_smash or can_use(Sticks) or can_use(Dins_Fire) or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Before Upper Lizalfos Pot 1": "
+                can_blast_or_smash or can_use(Sticks) or has_fire_source or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
+            "Dodongos Cavern MQ Before Upper Lizalfos Pot 2": "
+                can_blast_or_smash or can_use(Sticks) or has_fire_source or 
+                (is_adult and (logic_dc_jump or Hover_Boots or Progressive_Hookshot))",
```

Comment:
You can combine sticks and hasfiresource into hasfiresourcewithtorch. 

The blast/smash route is going to be reversing past the lizalfos, and so we'll need to make sure we have a weapon for that. Most ways of destroying those boulders can deal with the Lizalfos already (Din's avoids them, Hammer you're adult, and bombs are considered logical for fighting Lizals), but we don't consider bombchus logical for lizalfos anymore. Sticks are already accounted for with the other routes, so the blast/smash route just needs to add a check for adult/slingshot/bombs/ksword. I think that's all that's needed, had to think about this one for a bit.